standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/StatisticsTestUtils.java

```diff
@@ -0,0 +1,788 @@
+import org.junit.Assert;
+
+import java.util.Collections;
+
+public class StatisticsTestUtils {
+
+  public static final String HIVE_ENGINE = "hive";
+
+  private static final double epsilon = Double.MIN_VALUE;
+
+  private StatisticsTestUtils() {
+    throw new AssertionError("Suppress default constructor for non instantiation");
+  }
+
+  /*------------ TYPE-AGNOSTIC METHODS ------------*/
+  /**
+   * Methods creating column statistics for a given table and partition.
+   * @param data the statistics data
+   * @param tbl the target table
+   * @param column the target column
+   * @param partName the target partition
+   * @return column statistics for a given table and partition.
+   */
+  public static ColumnStatistics createColStats(ColumnStatisticsData data, Table tbl, FieldSchema column, String partName) {
+    ColumnStatisticsObj statObj = new ColumnStatisticsObj(column.getName(), column.getType(), data);
+    ColumnStatistics colStats = new ColumnStatistics();
+    ColumnStatisticsDesc statsDesc = new ColumnStatisticsDesc(true, tbl.getDbName(), tbl.getTableName());
+    statsDesc.setPartName(partName);
+    colStats.setStatsDesc(statsDesc);
+    colStats.setStatsObj(Collections.singletonList(statObj));
+    colStats.setEngine(HIVE_ENGINE);
+    return colStats;
+  }
+
+  /**
+   * Creates an FM sketch object initialized with the given values.
+   * @param values the values to be added
+   * @return an FM sketch initialized with the given values.
+   */
+  public static FMSketch createFMSketch(long... values) {
+    FMSketch fm = new FMSketch(1);
+    for (long value : values) {
+      fm.addToEstimator(value);
+    }
+    return fm;
+  }
+
+  /**
+   * Creates an FM sketch object initialized with the given values.
+   * @param values the values to be added
+   * @return an FM sketch initialized with the given values.
+   */
+  public static FMSketch createFMSketch(String... values) {
+    FMSketch fm = new FMSketch(1);
+    for (String value : values) {
+      fm.addToEstimator(value);
+    }
+    return fm;
+  }
+
+  /**
+   * Creates an HLL object initialized with the given values.
+   * @param values the values to be added
+   * @return an HLL object initialized with the given values.
+   */
+  public static HyperLogLog createHll(long ... values) {
+    HyperLogLog hll = HyperLogLog.builder().build();
+    for (long value : values) {
+      hll.addLong(value);
+    }
+    return hll;
+  }
+
+  /**
+   * Creates an HLL object initialized with the given values.
+   * @param values the values to be added
+   * @return an HLL object initialized with the given values.
+   */
+  public static HyperLogLog createHll(String ... values) {
+    HyperLogLog hll = HyperLogLog.builder().build();
+    for (String value : values) {
+      hll.addBytes(value.getBytes());
+    }
+    return hll;
+  }
+
+  private static void assertBitVector(Object hll, boolean isSetBitVectors, byte[] bitVector) {
```

Comment:
I am bit skeptical about all these `assertXX` utility methods. A lot of new code with multiple asserts and new messages. There are also a lot of similarities among the different variants. 

I was thinking that we could possibly call the plain `Assert.assertEquals` in the tests and pass the expected object (an instance of `ColumnStatisticsData`) with the actual object we get from the aggregator. If this is possible then we may not need to introduce these new utility methods.

Reply:
I have re-worked the tests to use plain `Assert.assertEquals` + the builder, it's much nicer indeed!